### ISSUE TITLE
are there plans to support eslint 9?

### ISSUE THREAD

**Issue by norflin321** [neoclide/coc-eslint#147]

**Comment by fannheyward** [comment]
Enable `eslint.experimental.useFlatConfig` and try again.

**Comment by Rom888** [comment]
I enabled eslint.experimental.useFlatConfig, but without success.
eslint shows errors from command line:
```
$ npx eslint ./**/config.service.ts

  24:26  error  Missing semicolon  @stylistic/semi
```

but no errors in vim.

I see no errors when :CocOpenLogs
```
2024-09-02T10:12:46.559 INFO (pid:13156) [plugin] - coc.nvim initialized with node: v22.7.0 after [33m160[39m
2024-09-02T10:13:01.127 INFO (pid:13156) [services] - LanguageClient ESLint state change: stopped => starting
2024-09-02T10:13:01.193 INFO (pid:13156) [services] - service tsserver started
2024-09-02T10:13:01.197 INFO (pid:13156) [language-client-index] - Language server "eslint" started with 13280
2024-09-02T10:13:01.273 INFO (pid:13156) [services] - LanguageClient ESLint state change: starting => running
2024-09-02T10:13:01.280 INFO (pid:13156) [services] - service eslint started
```

How can I troubleshoot it?

**Comment by Rom888** [comment]
Solved by removing package.json from the parent folder.
This helped:
https://github.com/neoclide/coc.nvim/wiki/Using-workspaceFolders
https://github.com/neoclide/coc.nvim/wiki/Debug-language-server#using-output-channel

**Comment by ravicious** [comment]
One pretty substantial inconvenience of `eslint.experimental.useFlatConfig` is that at work we have three release branches and a trunk. ESLint 9 is merged to trunk, but if I switch to a release branch, I have to set `eslint.experimental.useFlatConfig` to false, otherwise coc-eslint throws an error when I try to save a file.

**Comment by nashbridges** [comment]
@ravicious one solution is to use `:CocLocalConfig`. That however would require for you to check `.vim/coc-settings.json` into git.

**Comment by trygveaa** [comment]
> That however would require for you to check .vim/coc-settings.json into git.

Instead of checking it it, I've added `/.vim/coc-settings.json` to `.git/info/exclude` in the repos I have with eslint 9, so I don't have to bother my other team mates which don't use vim with this file.

**Comment by ravicious** [comment]
Right, but in the case that I described this is no different than just adjusting the global config. If you don't commit the local config to git (which I suppose you almost never want to do), then all release branches and trunk use the same coc settings, no matter whether the actual branch you're on has ESLint 9 or not.

**Comment by trygveaa** [comment]
Right, if you have branches with different versions it's definitely a problem. I forgot that you wrote about that.

It looks like the current implementation was pulled from an early version of the eslint 9 support in [vscode-eslint](https://github.com/microsoft/vscode-eslint). That extension now has support for automatically detecting if flat config should be used or not, so one could probably look at what that is doing.

**Comment by chemzqm** [comment]
Supported on coc-eslint@3.0.14
